python-threatexchange/threatexchange/api.py

```diff
@@ -483,13 +483,39 @@ def copy_threat_descriptor(self, postParams, showURLs, dryRun):
 
         return self.upload_threat_descriptor(postParams, showURLs, dryRun)
 
+    def delete_threat_descriptor(
```

Comment:
nit: can we add a note about how this actually requests the expiration of a td that would result in it no longer being visible in `threat_updates` and its eventual deletion? (I think that is what this is actually doing on the backend right?). We wouldn't want folks to misconstrue the capabilities of the API.

Reply:
I worry that DELETE is too ambiguous in the print statements etc.

Reply:
maybe just call it `expire_threat_descriptor` and mention that it should be deleted within x-time in the doc string?

Reply:
It doesn't expire the descriptor, it deletes the descriptor. What expires is the edge from collaboration to indicator.

I also was confused at this and had to look through the threat updates code to get a refresher but the descriptor is still deleted

Quick refresher:
1. When a threat descriptor expires it is deleted async
2. On all descriptor mutations (including deletes) the Collaboration -> Indicator edge is refreshed with a new update time and an expire time if necessary